src/redux/missions/MissionsSlice.js

```diff
@@ -0,0 +1,54 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// Initial State
+const initialState = [];
+
+// Base URL
+const url = 'https://api.spacexdata.com/v3/missions';
+
+// Async Action Creators
+export const fetchMissionsApi = createAsyncThunk(
+  'missions/fetchMissionsApi',
+  async () => {
+    const response = await axios.get(url);
+    return response.data;
+  },
+);
+
+// Slice Reducer
+const missionsSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    joinMission: {
+      reducer: (state, action) => state.map((item) => (
+        item.missionid !== action.payload ? item
+          : { ...item, missionjoin: true }
+      )),
+    },
+    unjoinMission: {
+      reducer: (state, action) => state.map((item) => (
+        item.missionid !== action.payload ? item
+          : { ...item, missionjoin: false }
+      )),
+    },
+  },
+  extraReducers: {
+    [fetchMissionsApi.fulfilled]: (state, action) => {
+      const missions = action.payload.map((el) => ({
+        // distructured data
+        missionid: el.mission_id,
+        missionName: el.mission_name,
+        missionDesc: el.description,
+        missionweblink: el.wikipedia,
+        missionjoin: false,
+      }));
+      return missions;
+    },
+  },
+});
```

Comment:
- [ ] Kindly use `Jest` to write unit tests for your reducers and other pure functions. 👍

Reply:
Done in commit id: https://github.com/mavericks-db/space-travelers-hub/pull/35/commits/170572e42c8d85200f5639e9dfee855a94953941